dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/ArtifactProblem.java

```diff
@@ -35,7 +35,7 @@
 
   protected final ImmutableList<DependencyNode> dependencyPath;
 
-  protected ArtifactProblem(Artifact artifact, List<DependencyNode> dependencyPath) {
+  protected ArtifactProblem(Artifact artifact, Collection<DependencyNode> dependencyPath) {
```

Comment:
As we care about the order of the dependency path, List is appropriate.

Reply:
OK, fixed, though stacks being a subclass of List is a long-recognized problem in the Java Collections API since it violates the ADT for stack.